### DISCUSSION TITLE
Understanding the computation of the Karcher median

### DISCUSSION THREAD

**Discussion by glegarda** [jdtuck/fdasrsf_python#65] [Q&A]
Hello,

I am trying to understand how the Karcher **median** is calculated in the alignment of a collection of functions. I went over the implementation of the `srsf_align()` function, but I'm not sure I understand where the steps taken [here](https://github.com/jdtuck/fdasrsf_python/blob/c939fee8f763965544d9b436de993d40f8df2ea7/fdasrsf/time_warping.py#L277) are coming from:
```
# Minimization Step
# compute the mean of the matched function
stp = 0.3
vbar = vtil.sum(axis=1) * (1 / dtil.sum())
qtemp = q[:, :, r + 1]
ftemp = f[:, :, r + 1]
mq[:, r + 1] = mq[:, r] + stp * vbar
```
Could you share some references that would help me get a better understanding of this bit?

Thanks!

**Comment by jdtuck** [comment]
Please look at the Srivastava and Klassen's book from 2016 entitled Functional and Shape Data Analysis.
